src/base/string_list.hpp

```diff
@@ -201,20 +223,46 @@ class string_list_t {
 private:
   static std::string escape_arg(const std::string& arg) {
     std::string escaped_arg;
+    bool needs_quotes = false;
+
+#ifdef _WIN32
+    // These escaping rules try to match parsing rules of Windows programs, e.g. as outlined here: 
+    // http://www.windowsinspired.com/how-a-windows-programs-splits-its-command-line-into-individual-arguments/
+    // "[..] backslashes are interpreted literally (except when they precede a double quote character)".
+    int backslashes = 0;
+    for (auto c : arg) {
+      if (c == '\\') {
+        ++backslashes;
+      } else { 
+        if (c == '\"') {
+          for (int i = 0; i < backslashes + 1; ++i) {
```

Comment:
nit: why not i <= backslashes?

Reply:
https://github.com/shapr3d/buildcache/pull/2/commits/1ec4e8eb6ddac9330daa2b38d31d0e9b0a29515e